### PR TITLE
Add the cmake macro to generate tests

### PR DESCRIPTION
**Pull request by Barthelemy**
It uses CTest and boost.test.
There is also an example in ExampleModule1.

### CODE REVIEW

File: cmake/O2Utils.cmake [hunk 1]
```diff
@@ -320,14 +321,55 @@ function(O2_GENERATE_EXECUTABLE)
       MODULE_LIBRARY_NAME ${PARSED_ARGS_MODULE_LIBRARY_NAME}
   )
 
-  ############### install the executable #################
-  install(TARGETS ${PARSED_ARGS_EXE_NAME} DESTINATION bin)
+  if (NOT ${PARSED_ARGS_INSTALL} OR ${PARSED_ARGS_INSTALL})
```
Comment (matthiasrichter) [inline cmake/O2Utils.cmake]: @Barthelemy while trying to find the reason for https://github.com/AliceO2Group/AliceO2/issues/196 I stumbled over this line. Isn't that a tautology? What was your intent?

Comment (Barthelemy) [inline cmake/O2Utils.cmake]: Good catch. It is a mistake. It is needed by GENERATE_TEST to tell GENERATE_EXECUTABLE not to install the tests. I will fix it.

Comment (Barthelemy) [inline cmake/O2Utils.cmake]: https://github.com/AliceO2Group/AliceO2/pull/201

**Comment by matthiasrichter** [comment]
Ah, this is actually partly handling my question about compilation and installation of test programs I've circulated two weeks ago. Good to know that you have though about this already.

**Comment by Barthelemy** [comment]
yes, sorry about being silent, I was very busy with Run 2. I will now be more active for O2.

**Pull request merged**